Added Double Circular Linked List

Aditya-y9 opened the pull request:
Honourable maintainers,
Please add Hacktoberfest and Hacktoberfest-accepted labels.


- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

susanhsrestha commented on src/main/java/com/thealgorithms/datastructures/lists/DoubleCircularLinkedList:
```diff
@@ -0,0 +1,134 @@
+/**
```
The Java file extension is missing, and the package naming convention is also not provided.

Aditya-y9 replied:
Sorry for that silly mistake i just overlooked it.
It has been fixed now.
Thanks!

Aditya-y9 commented:
@susanhsrestha @BurkHufnagel @arindam89 @qckzr 
Done with all checks
Added Java file extension
Added package accordingly
Please Add Hacktoberfest and Hacktoberfest-accepted labels

Thanks!

susanhsrestha commented:
All I see is an empty file.

susanhsrestha reviewed (approved):
LGTM!
